Bottom: e095de5d3ae86aa69b264ca781aa956a9adc1d42
Top:    6ec84e1a4c585c92226630339ff2bb21599e2836
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 17:19:47 +0800

Refresh of autofs-5.0.7-fix-several-off-by-one-errors.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index 67ea525..c1fc528 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1135,10 +1135,10 @@ static int mount_subtree(struct autofs_point *ap, struct mapent *me,
 			}
 			ro_len = strlen(ro_loc);
 
-			tmp = alloca(mnt_root_len + 1);
+			tmp = alloca(mnt_root_len + 2);
 			strcpy(tmp, mnt_root);
-			tmp[mnt_root_len - 1] = '/';
-			tmp[mnt_root_len] = '\0';
+			tmp[mnt_root_len] = '/';
+			tmp[mnt_root_len + 1] = '\0';
 			root = tmp;
 
 			rv = sun_mount(ap, root, name, namelen, ro_loc, ro_len, myoptions, ctxt);
